### PR TITLE
ci: update basic-workflow-tests.sh to work with vllm

### PR DESCRIPTION
**Pull request by alimaredia**
**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by alimaredia** [comment]
Blocked by: https://github.com/instructlab/instructlab/pull/1839

File: scripts/basic-workflow-tests.sh [hunk 1]
```diff
@@ -120,8 +121,8 @@ test_init() {
         ilab config init --non-interactive
     fi
     step Checking config.yaml
-    if [ "${MIXTRAL}" -eq 1 ]; then
-        sed -i -e 's|merlinite.*|mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf|' "${CONFIG_HOME}/instructlab/config.yaml"
+    if [ "${MIXTRAL}" -eq 1 ] && [ "${BACKEND}" == "vllm" ]; then
```
Comment (booxter) [inline scripts/basic-workflow-tests.sh:124]: `set_defaults` seems to guarantee that it's never `!= "vllm"`. Wonder if we have to check it everywhere below.

File: scripts/basic-workflow-tests.sh [hunk 2]
```diff
@@ -120,8 +121,8 @@ test_init() {
...
+        sed -i -e 's|merlinite.*|${MIXTRAL_GPTQ_MODEL}|' "${CONFIG_HOME}/instructlab/config.yaml"
```
Comment (booxter) [inline scripts/basic-workflow-tests.sh:125]: This won't work: `''` won't expand the variable

**Review by booxter — CHANGES_REQUESTED** [review]

**Review by booxter — APPROVED** [review]
LGTM. Some of `if MIXTRAL && vllm` checks could probably be just `if MIXTRAL` because `set_defaults` guarantee that it's never one but not another.

**Comment by alimaredia** [comment]
@booxter When I started this line of work I had vllm and non-vllm cases when flags for granite, mixtral, and neither were passed in. When Jason and I talked yesterday we decided that this was overkill.

The reason I left the `if Mixtral && vllm` checks were just so that it was obvious to someone new to the script that the two are intertwined. That being said I removed them, since they were somewhat of a relic of past work I'd done.

**Review by booxter — COMMENTED** [review]
Code-wise, looks correct. Which CI job utilizes the workflow?

**Review by leseb — APPROVED** [review]

**Comment by leseb** [comment]
I tried to run the CI job manually but it failed [here](https://github.com/instructlab/instructlab/actions/runs/10074501141/job/27850757333)

**Pull request closed without merging**